### PR TITLE
Don't clear mid property if stopped.

### PR DESCRIPTION
**Pull request by jan-ivar**
Fixes https://github.com/w3c/webrtc-pc/issues/2238. @docfaraday @henbos PTAL.

***
<a href="https://pr-preview.s3.amazonaws.com/jan-ivar/webrtc-pc/pull/2239.html" title="Last updated on Jul 26, 2019, 6:59 PM UTC (e9357bc)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/webrtc-pc/2239/72ed617...jan-ivar:e9357bc.html" title="Last updated on Jul 26, 2019, 6:59 PM UTC (e9357bc)">Diff</a>

### CODE REVIEW

**Comment by jan-ivar** [comment]
This is a JSEP mistake we're fixing in webrtc-pc basically.

File: webrtc.html [hunk 1]
```diff
@@ -7752,10 +7752,17 @@ <h3><dfn>RTCRtpTransceiver</dfn> Interface</h3>
...
-      description.</p>
+      a <a>media description</a> if its "mid property" is non-null; otherwise it is
+      said to be <dfn>disassociated</dfn>.
+      Conceptually, a transceiver becomes <a>associated</a> once it has been
+      successfully represented in a session description. Once associated, a
+      transceiver only becomes <a>disassociated</a> if it is immediately rolled
+      back, or if the transceiver has been <a>stopped</a> and its m-line has
+      been recycled, as defined in <span data-jsep=
+      "subsequent-offers subsequent-answers">[[!JSEP]]</span>. In the latter
+      case, the <code><a>RTCRtpReceiver</a></code>'s
```
Comment (henbos) [inline webrtc.html]: the RTCRtpTransceiver's mid value

File: webrtc.html [hunk 2]
```diff
@@ -7778,6 +7785,10 @@ <h3><dfn>RTCRtpTransceiver</dfn> Interface</h3>
           <p>Let <var>transceiver</var> have a <dfn>[[\Receiver]]</dfn> internal
           slot, initialized to <var>receiver</var>.</p>
         </li>
+        <li data-tests="RTCPeerConnection-addTransceiver.https.html">
+          <p>Let <var>transceiver</var> have a <dfn>[[\Mid]]</dfn> internal
+          slot, initialized to <var>null</var>.</p>
+        </li>
```
Comment (henbos) [inline webrtc.html]: An internal slot is good. But we have existing language (i.e. when setting an RTCSessionDescription) that says to "Set transceiver's mid value to the mid of the media description" and similar sentences in several places. Update those to modify the internal slot.

File: webrtc.html [hunk 3]
```diff
@@ -7841,6 +7852,28 @@ <h2>Attributes</h2>
               negotiation is complete, the <code>mid</code> value may be null.
               After rollbacks, the value may change from a non-null value
               to null.</p>
+              <p>On getting, the attribute MUST return the value of the
+              <a>[[\Mid]]</a> slot.</p>
+              <p>Whenever <span data-jsep=
+              "subsequent-offers subsequent-answers">[[!JSEP]]</span> says to
+              modify the "mid property", the user agent MUST queue a task that
+              runs the following steps:</p>
```
Comment (henbos) [inline webrtc.html]: I think this is the wrong place to put this algorithm. This decouples it from setting an RTCSessionDescription and it becomes unclear whether to update the mid before or after firing events etc. But is this algorithm even needed, we're already updating the mid inside of the task that was queued for SLD/SRD. Let's not queue two tasks.

File: webrtc.html [hunk 4]
```diff
@@ -7752,10 +7752,17 @@ <h3><dfn>RTCRtpTransceiver</dfn> Interface</h3>
...
-      property is non-null; otherwise it is said to be disassociated. Conceptually, an
-      <a>associated</a> transceiver is one that's represented in the last applied session
-      description.</p>
+      a <a>media description</a> if its "mid property" is non-null; otherwise it is
+      said to be <dfn>disassociated</dfn>.
+      Conceptually, a transceiver becomes <a>associated</a> once it has been
+      successfully represented in a session description. Once associated, a
+      transceiver only becomes <a>disassociated</a> if it is immediately rolled
+      back, or if the transceiver has been <a>stopped</a> and its m-line has
+      been recycled, as defined in <span data-jsep=
```
Comment (henbos) [inline webrtc.html]: I think the "if the transceiver has been stopped and its m-line has been recycled" should be part of the SLD/SRD steps.

Comment (aboba) [reply webrtc.html]: Why remove the link from "mid"?  Also, "successfully represented in a session description" is not as specific as "last applied session description".

**Pull request closed without merging**